### PR TITLE
Issue #1232; Replaced usage of var to const/let in some files

### PR DESCRIPTION
**Pull request by SangameshItagi**
Addresses some files referenced in issue https://github.com/projectmesa/mesa/issues/1232

Replaced var with const/let in following files:

    mesa/visualization/templates/js/CanvasHexModule.js
    mesa/visualization/templates/js/CanvasModule.js
    mesa/visualization/templates/js/GridDraw.js
    mesa/visualization/templates/js/HexDraw.js

I have tried to use const wherever possible. Please comment or let me know if any modification is required. Thanks!

### CODE REVIEW

**Comment by rht** [comment]
I checked `mesa/visualization/templates/js/CanvasHexModule.js`. Every `var` in that file needs to be replaced with `const`, because none of the variables are being reassigned with the `=` operator. Let me know once you have used this rule in the other files, and I will review again.

**Comment by SangameshItagi** [comment]
Hey @rht , Thanks for taking time and reviewing it. I have edited the PR with the changes you suggested in review and also suggestions in the other channel running with @trickroll. Please let me know if anymore suggestions that needs be included in this PR.
Thanks!

File: mesa/visualization/templates/js/GridDraw.js [hunk 1]
```diff
@@ -260,8 +260,6 @@ var GridVisualization = function(width, height, gridWidth, gridHeight, context,
 
                 // This part draws the text inside the ArrowHead
                 if (text !== undefined) {
-                        var cx = (x + 0.5) * cellWidth;
```
Comment (rht) [inline mesa/visualization/templates/js/GridDraw.js:263]: OK, these 2 lines are redundant declaration.

File: mesa/visualization/templates/js/HexDraw.js [hunk 2]
```diff
@@ -131,24 +132,25 @@ var HexVisualization = function(width, height, gridWidth, gridHeight, context, i
...
+		    const angle_rad = Math.PI / 180 * angle_deg
 		    return [(x + size * Math.cos(angle_rad) * 1.2),
 		            (y + size * Math.sin(angle_rad))]
 		}
 
 
-		var px, py
+		let px, py
 		context.beginPath();
 		[px, py] = hex_corner(cx,cy,r,1)
```
Comment (rht) [inline mesa/visualization/templates/js/HexDraw.js:155]: `const [px, py] = hext_corner(...)`

Comment (rht) [reply mesa/visualization/templates/js/HexDraw.js:155]: Oh, `[px, py]` are redeclared later. In thase case, `let [px, py] = hex_corner(...)`

**Comment by rht** [comment]
2 remaining things:
- https://github.com/projectmesa/mesa/pull/1248/files#r843296813=
- Use this as the commit message: "refactor: Replace usage of var with const/let in some JS files", otherwise it is too long. Usually, the issue is referenced in the commit message body instead.

File: mesa/visualization/templates/js/HexDraw.js [hunk 3]
```diff
@@ -131,26 +132,26 @@ var HexVisualization = function(width, height, gridWidth, gridHeight, context, i
...
+		    const angle_rad = Math.PI / 180 * angle_deg
 		    return [(x + size * Math.cos(angle_rad) * 1.2),
 		            (y + size * Math.sin(angle_rad))]
 		}
 
 
-		var px, py
 		context.beginPath();
-		[px, py] = hex_corner(cx,cy,r,1)
+		let [px, py] = hex_corner(cx,cy,r,1)
```
Comment (SangameshItagi) [inline mesa/visualization/templates/js/HexDraw.js:154]: @rht fixed it as suggested by you

**Comment by SangameshItagi** [comment]
@rht Have fixed the above 2 issues as suggested. Thanks!

**Comment by rht** [comment]
LGTM

**Pull request merged**